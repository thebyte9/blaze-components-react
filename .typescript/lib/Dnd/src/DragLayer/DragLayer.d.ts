import React from "react";
interface IDragLayerProps {
    dragItem: any;
    dragLayerRef: any;
}
declare const DragLayer: React.SFC<IDragLayerProps>;
export default DragLayer;
