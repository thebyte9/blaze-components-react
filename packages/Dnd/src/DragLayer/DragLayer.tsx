import React from "react";
interface IDragLayerProps {
  dragItem: any;
  dragLayerRef: any;
}
const DragLayer: React.SFC<IDragLayerProps> = ({
  dragItem,
  children,
  dragLayerRef
}) => {
  const el: any = document.getElementById(dragItem.id);
  return (
    <div className="nestable-drag-layer">
      <ol
        className="nestable-list"
        ref={dragLayerRef}
        style={{ width: el.clientWidth }}
      >
        {children}
      </ol>
    </div>
  );
};
export default DragLayer;
