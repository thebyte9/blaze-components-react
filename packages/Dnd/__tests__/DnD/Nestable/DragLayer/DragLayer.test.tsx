import { cleanup, render } from "@testing-library/react";
import React from "react";
import DragLayer from "../../../../src/DragLayer";
import { dragLayerProps } from "../../../mocks/dnd";

describe("Drag layer", () => {
  afterEach(cleanup);

  it("should be defined", () => {
    expect(DragLayer).toBeDefined();
  });

  it("should render without throwing error", () => {
    document.getElementById = jest.fn();
    const { mockReturnValue }: any = document.getElementById;
    mockReturnValue({ clientWidth: 300 });
    const { container } = render(<DragLayer {...dragLayerProps} />);
    expect(container).toMatchSnapshot();
  });
});
