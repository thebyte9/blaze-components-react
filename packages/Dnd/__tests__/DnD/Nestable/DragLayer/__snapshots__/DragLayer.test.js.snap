// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drag layer should render without throwing error 1`] = `
<div>
  <div
    class="nestable-drag-layer"
  >
    <ol
      class="nestable-list"
      style="width: 300px;"
    >
      <div>
        children
      </div>
    </ol>
  </div>
</div>
`;
